src/test/regress/expected/pg15.out

```diff
@@ -984,6 +984,151 @@ select min(x), max(x) from xid8_t1 GROUP BY y ORDER BY 1,2;
+SELECT relname, reloptions FROM pg_class
+WHERE relname = 'sec_invoker_view' AND relnamespace = 'pg15'::regnamespace;
+     relname      |       reloptions
+---------------------------------------------------------------------
+ sec_invoker_view | {security_invoker=true}
+(1 row)
+
+\c - - - :master_port
+SET search_path TO pg15;
+-- test altering the security_invoker flag
+ALTER VIEW sec_invoker_view SET (security_invoker = false);
+\c - - - :worker_1_port
+SELECT relname, reloptions FROM pg_class
+WHERE relname = 'sec_invoker_view' AND relnamespace = 'pg15'::regnamespace;
+     relname      |        reloptions
+---------------------------------------------------------------------
+ sec_invoker_view | {security_invoker=false}
+(1 row)
+
+\c - - - :master_port
+SET search_path TO pg15;
+ALTER VIEW sec_invoker_view SET (security_invoker = true);
+-- create a new user but don't give select permission to events table
+-- only give select permission to the view
+CREATE ROLE rls_tenant_1 WITH LOGIN;
+GRANT USAGE ON SCHEMA pg15 TO rls_tenant_1;
+GRANT SELECT ON sec_invoker_view TO rls_tenant_1;
+-- this user shouldn't be able to query the view
+-- because the view is security invoker
+-- which means it will check the invoker's rights
+-- against the view's underlying tables
+SET ROLE rls_tenant_1;
+SELECT * FROM sec_invoker_view ORDER BY event_id;
+ERROR:  permission denied for table events
+RESET ROLE;
+-- now grant select on the underlying distributed table
+-- and try again
+-- now it should work!
+GRANT SELECT ON TABLE events TO rls_tenant_1;
+SET ROLE rls_tenant_1;
+SELECT * FROM sec_invoker_view ORDER BY event_id;
+ tenant_id | event_id | descr
+---------------------------------------------------------------------
+         1 |        1 | push
+         2 |        2 | push
+(2 rows)
+
+RESET ROLE;
+-- Enable row level security
+ALTER TABLE events ENABLE ROW LEVEL SECURITY;
+-- Create policy for tenants to read access their own rows
+CREATE POLICY user_mod ON events
+  FOR SELECT TO rls_tenant_1
+  USING (current_user = 'rls_tenant_' || tenant_id::text);
+-- all rows should be visible because we are querying with
+-- the table owner user now
+SELECT * FROM sec_invoker_view ORDER BY event_id;
+ tenant_id | event_id | descr
+---------------------------------------------------------------------
+         1 |        1 | push
+         2 |        2 | push
+(2 rows)
+
+-- Switch user that has been granted rights,
+-- should be able to see rows that the policy allows
+SET ROLE rls_tenant_1;
+SELECT * FROM sec_invoker_view ORDER BY event_id;
+ tenant_id | event_id | descr
+---------------------------------------------------------------------
+         1 |        1 | push
+(1 row)
+
+RESET ROLE;
+-- ordinary view on top of security invoker view permissions
+-- ordinary means security definer view
+-- The PG expected behavior is that this doesn't change anything!!!
+-- Can't escape security invoker views by defining a security definer view on top of it!
```

Comment:
This is an interesting case, thanks for adding a test. What about the order is changed, on the outer we have security invoker view and inside security definer. Do we still get errors?

Reply:
Yes, we still get errors as if it were a security definer view (like the errors in the issue). I will not add a test here because we haven't added tests for "failing" security definer views. I will add this case in the issue.